src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/prior_box/prior_box_kernel_ref.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "prior_box_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+#include <iostream>
+
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData SetDefault(const prior_box_params& params, const optional_params&) {
+    kernel_selector::CommonDispatchData dispatchData;
+    dispatchData.gws = {params.width, params.height, 1};
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    return dispatchData;
+}
+}  // namespace
+
+KernelsData PriorBoxKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    KernelData kernel_data = KernelData::Default<prior_box_params>(params);
+    const prior_box_params& new_params = dynamic_cast<const prior_box_params&>(*kernel_data.params.get());
+
+    auto dispatch_data = SetDefault(new_params, options);
+    auto entry_point = GetEntryPoint(kernelName, new_params.layerID, params, options);
+    auto specific_jit = GetJitConstants(new_params);
+    auto jit = CreateJit(kernelName, specific_jit, entry_point);
+    FillCLKernelData(kernel_data.kernels[0],
+                     dispatch_data,
+                     params.engineInfo,
+                     kernelName,
+                     jit,
+                     entry_point,
+                     "",
+                     false,
+                     false,
+                     2);
+
+    KernelsData kernelsData;
+    kernelsData.push_back(std::move(kernel_data));
+    return kernelsData;
+}
+
+KernelsPriority PriorBoxKernelRef::GetKernelsPriority(const Params& /*params*/,
+                                                      const optional_params& /*options*/) const {
+    return FORCE_PRIORITY_1;
+}
+
+ParamsKey PriorBoxKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableDifferentTypes();
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+    k.EnableBatching();
+    return k;
+}
+
+bool PriorBoxKernelRef::Validate(const Params& params, const optional_params& optionalParams) const {
+    if (params.GetType() != KernelType::PRIOR_BOX || optionalParams.GetType() != KernelType::PRIOR_BOX) {
+        return false;
+    }
+
+    // output size, image size
+    constexpr uint32_t number_of_inputs = 2;
+    auto& priorBox8arams = dynamic_cast<const prior_box_params&>(params);
+    if (priorBox8arams.inputs.size() != number_of_inputs) {
+        return false;
+    }
+
+    return true;
+}
+
+JitConstants PriorBoxKernelRef::GetJitConstants(const prior_box_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    const std::string name = "PRIOR_BOX_";
```

Comment:
I think we do not need this prefix except those identifiers which overlaps with default JIT ones - if there are any. It just makes kernel code more verbose.